frontend/src/modules/EditZing/Components/StudentCard.tsx

```diff
@@ -33,6 +36,21 @@ const StudentCard = ({
     setSelected(!selected)
   }
 
+  const [openNotes, setOpenNotes] = useState(false)
+  const [saved, setSaved] = useState(false)
+  const [notSaved, setNotSaved] = useState(false)
+  const handleOpenNotes = () => setOpenNotes(true)
+  const handleCloseNotes = () => setOpenNotes(false)
+
+  const [studentNotes, setStudentNotes] = useState('')
+
+  useEffect(() => {
+    const savedNote = student.groups.find(
+      (g: GroupMembership) => g.courseId === courseId
+    )?.notes
+    setStudentNotes(savedNote || '')
+  }, [student, courseId])
```

Comment:
I actually don't think keeping student notes as local state for the StudentCard is a good idea because it will be out of sync with the students the top level component sees. If we were to use notes elsewhere on this page, the changes here would not be reflected. I would suggest lifting the note-updating logic up to the top and passing down a function to update the notes for a particular student.

Reply:
Okay, I reverted it back to a single state. However, this method has the effect of showing the filled notes icon when closing but not explicitly saving. It also saves the value of the canceled note until refreshed. One potential solution I could try is to pass in two different closing methods, one that simply sets the note state back to what it originally was if the user doesn't save, or just handle close regularly if the user saves. Or, just leave it as is?

Reply:
See new review comment